examples/account/key_exchange.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Demonstrates Elliptic-curve Diffie-Hellman (ECDH) cryptographic key exchange with DID Documents.
+//!
+//! cargo run --example key_exchange
+
+use std::path::PathBuf;
+
+use identity::account::Account;
+use identity::account::AccountBuilder;
+use identity::account::EncryptionKey;
+use identity::account::IdentitySetup;
+use identity::account::MethodContent;
+use identity::account::Result;
+use identity::account_storage::EncryptedData;
+use identity::account_storage::Stronghold;
+use identity::did::MethodScope;
+use identity::iota::Client;
+use identity::iota::ResolvedIotaDocument;
+use identity::iota::TangleResolve;
+use identity::iota_core::IotaVerificationMethod;
+
+mod create_did;
+
+pub async fn run() -> Result<()> {
+  // Alice and Bob want to communicate securely by encrypting their messages so only they
+  // can read them. They both publish DID Documents with X25519 public keys and use them
+  // to derive a shared secret key for encryption.
+
+  // Sets the location and password for the Stronghold
+  //
+  // Stronghold is an encrypted file that manages private keys.
+  // It implements best practices for security and is the recommended way of handling private keys.
+  let stronghold_path: PathBuf = "./example-strong.hodl".into();
+  let password: String = "my-password".to_owned();
+  let stronghold: Stronghold = Stronghold::new(&stronghold_path, password, None).await?;
+
+  // Alice creates and publishes their DID Document (see create_did and manipulate_did examples).
+  let mut builder: AccountBuilder = Account::builder().storage(stronghold);
+  let mut alice_account: Account = builder.create_identity(IdentitySetup::default()).await?;
+  alice_account
+    .update_identity()
+    .create_method()
+    .content(MethodContent::GenerateX25519)
+    .fragment("kex-0")
+    .scope(MethodScope::key_agreement())
+    .apply()
+    .await?;
+
+  // Bob creates and publishes their DID Document (see create_did and manipulate_did examples).
+  let mut bob_account: Account = builder.create_identity(IdentitySetup::default()).await?;
+  bob_account
+    .update_identity()
+    .create_method()
+    .content(MethodContent::GenerateX25519)
+    .fragment("kex-0")
+    .scope(MethodScope::key_agreement())
+    .apply()
+    .await?;
+
+  // Alice and Bob tell each other their DIDs. They each resolve the DID Document of the other
+  // to obtain their X25519 public key. Note that in practice, they would run this code completely
+  // separately.
+  let client: Client = Client::new().await?;
```

Comment:
I am wondering whether the `Resolver` is more fitting than the `Client` here as the only Client related operations performed is DID Document resolution. Maybe @cycraig has an opinion on this matter?